test/fixtures/api_actions.yml

```diff
@@ -124,3 +124,90 @@ create_api_action_plugin_bishop_monitoring_email:
   include_api_resource_data: true
   email: test@restarone.com
   api_namespace: monitoring_target_incident
+
+custom_api_action_on_timesheet_request:
+  type: CreateApiAction
+  action_type: custom_action
+  include_api_resource_data: true
+  payload_mapping:
+  api_namespace: timesheet_request
+  method_definition: |
+    # get time tracker namespace
+    time_tracker = ApiNamespace.find_by(name: 'time_tracker')
+    
+    format = "%Y-%m-%d"
+    if api_resource.properties['current_month'] == "true"
+      api_resource.update(properties: api_resource.properties.merge({ start_date: Date.today.beginning_of_month.strftime(format), end_date: Date.today.end_of_month.strftime(format) }))
+    end
+    
+    # get time tracker entries for the current user within the specified time
+    start_date = Date.parse(api_resource.properties['start_date'])
+    end_date = Date.parse(api_resource.properties['end_date'])
+    # email = current_user.email
```

Comment:
@atosh502 have you tested this? Let me know what happens when this code runs

Reply:
yes @donrestarone I ran the tests and it was successful.... the actual code deployed at team.restarone.solutions doesn't have this line commented.... I created the tests and the fixtures specifically for the user with a specific email to test this

Reply:
@atosh502 You can set the `current_user` as shown below for api_actions. [LINK](https://github.com/restarone/violet_rails/pull/1204/files#diff-0a7f1bdacef9aafc31b98b5e1a3ca1c2bdca203795e544e79275dd01f1f16025R12-R13)
```
Current.user = users(:one)
```

You can further see the implementation of `custom_api_action` for clarification.
[LINK](https://github.com/atosh502/violet_rails/blob/master/app/models/api_action.rb#L78-L91)